### PR TITLE
Normalization of elevation and azimuth angles for surface plots

### PR DESCRIPTION
**Pull request by jinshifen33**
## PR Summary
This PR normalizes the input during the initialization of a surface plot in order to make the elevation and azimuth angles consistent throughout the plot.

[Issue #10241](https://github.com/matplotlib/matplotlib/issues/10241)

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant

### CODE REVIEW

**Comment by WeatherGod** [comment]
I need to think on this a bit. The original code intentionally flipped the display upside down, but I haven't a clue why anyone would want that. But, just because I can't think of a reason right now doesn't mean that it isn't supposed to be this way.

**Comment by jinshifen33** [comment]
Have looked at the code as well, it was like that since 2009 ..

**Comment by jinshifen33** [comment]
I found a use case, at least, that will make this fix useful.

From https://matplotlib.org/examples/mplot3d/rotate_axes3d_demo.html 

With changing Elevation angle from (0-360) as an example.

In the current code base, the rotation will be discontinued at 270(i.e  the 3d plot got flipped by 180deg) (see below)

orig


With the fixed code , it will complete one whole rotation(see the gif provided)  

fixed

**Comment by jinshifen33** [comment]
If you referring to the flipping behaviour of the variable V, then I think it is a normal behaviour. The fix did not attempt to change that, the fix only normalize the input angle on initialization of the view, to make it consistent throughout the plotting. As you might notice, it is not only the plot got flipped on z-direction 180 degree, the azimuth also got flipped on the original bug report.

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 1]
```diff
@@ -964,16 +964,16 @@ def view_init(self, elev=None, azim=None):
...
 
         self.dist = 10
-
+        # normalize the angle to stay between [-180, 180]
         if elev is None:
-            self.elev = self.initial_elev
+            self.elev = art3d.norm_angle(self.initial_elev)
         else:
-            self.elev = elev
+            self.elev = art3d.norm_angle(elev)
```
Comment (eric-wieser) [inline lib/mpl_toolkits/mplot3d/axes3d.py]: If you're going to normalize it at all, elevation should be normalized to `[-90, +90]`

Comment (jinshifen33) [reply lib/mpl_toolkits/mplot3d/axes3d.py]: looking at code here: https://github.com/jinshifen33/matplotlib/blob/master/lib/mpl_toolkits/mplot3d/axes3d.py#L1214  , the angle is kept in [-180, 180]

**Comment by jinshifen33** [comment]
hey @WeatherGod  , have you made up your mind?

**Comment by WeatherGod** [comment]
you  know what? That animation convinced me. I'll approve.

**Comment by eric-wieser** [comment]
This still strikes me as the wrong fix. We should be normalizing it when we render, not when we store - there might be graphs that want to do something clever with the view angles, and expect them to be continuous.

**Comment by jinshifen33** [comment]
it seems to me view_init is a method elevation and azimuth used in rendering.(they have to be stored some where)

if there are need for raw angle operation (non normalized) on elev and azimu, then have it done by implementing  getter and setter on these two attributes.

and when start drawing, a call to view_init should normalize the operated angles.

And i dont understand what do you mean by "continuous" ..

**Comment by jinshifen33** [comment]
@eric-wieser  any comment ?

**Comment by jklymak** [comment]
Re milestoning 3.1 as this looks to need more discussion....

**Comment by scottshambaugh** [comment]
This PR was subsumed into https://github.com/matplotlib/matplotlib/pull/21426 and can be closed.

**Pull request closed without merging**